contacts/views.py

```diff
@@ -0,0 +1,89 @@
+from django.db.utils import IntegrityError
+from rest_framework import generics, permissions, status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+
+from .models import Case, Contact
+from .serializers import CaseSerializer, ContactSerializer
+
+
+class ConfirmedContactView(generics.GenericAPIView):
+    """
+    POST endpoint to create a case and contact.
+    ---
+    POST fields:
+    * msisdn
+    * timestamp
+    * external_id
+    - name (optional)
+    - case_id (optional)
+    ---
+    Response example:
+    ```
+    {
+        "id": "84fe2a6c-a5f8-4994-9d96-d765514740b3",
+        "msisdn": "+27820001001",
+        "date_start": "2020-07-19T19:35:11.400070Z",
+        "case_id": null,
+        "name": null,
+        "created_by": "admin",
+        "created_at": "2020-07-21T13:09:06.921305Z",
+        "external_id": "51e7f725b59442c7b75b48752b29f243"
+    }
+    ```
+    """
+
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+        data = request.data
+
+        msisdn = data.pop("msisdn", None)
+
+        # Try to create a
+        try:
+            contact_serializer = ContactSerializer(data={"msisdn": msisdn})
+            contact_serializer.is_valid(raise_exception=True)
+            contact = contact_serializer.save()
+        except (IntegrityError, AssertionError, ValidationError):
```

Comment:
I'm not sure when we would get an `AssertionError`, but do we now want the `ValidationError` to be uncaught?

eg. if they don't supply the `msisdn` field, then we want the `contact_serializer.is_valid(raise_exception=True)` to raise an exception, which will result in a 400 with a descriptive error message being sent.

Reply:
ValidationError is raising different error message - I replace it with custom message.

Reply:
Same for `ValidationError` in Case. The only difference is that in Case it can be skipped since the custom message is nothing special - but in Contact catching unique fail means we have to fetch an existing Contact. Custom errors are raised with `APIException` and are not caught by this exception handler.